adapters/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/envelope/RestEnvelopeBuilder.java

```diff
@@ -101,8 +102,17 @@ private JsonObject buildPayload() {
                 .map(JsonObjects::createObjectBuilder)
                 .orElse(Json.createObjectBuilder());
 
-        for (Map.Entry<String, String> entry : params.orElse(emptyMap()).entrySet()) {
-            payloadBuilder = payloadBuilder.add(entry.getKey(), entry.getValue());
+        for (Parameter param : params.orElse(emptyList())) {
```

Comment:
Do you need the orElse here? would params ever be null? and if it is an emptyList then the foreach loop will just skip.